src/components/FeatureViews/Search/SearchResults.js

```diff
@@ -156,31 +71,72 @@ const SearchResults = (props) => {
 
     if ((searchState.searchQueryStatus.account !== 200 ||
         searchState.searchQueryStatus.course !== 200) && searchState.searchQueryStatus.status) {
-        return <Loading />
+        return <SearchResultsLoader
+            SearchResults={numberOfResults()}
+            accountPage={currentPage.account}
+            coursePage={currentPage.course}
+            SearchQuery={SearchQuery}/>
     }
 
     const MAX_PAGE = Math.ceil(numberOfResults() / 8);
-    const pageNumber = (apiPage, startPage) => {
-        const firstResultsPage = startPage === 0 && apiPage == 1;
-        const secondResultsPage = startPage !== 0 && apiPage == 1;
-        if (firstResultsPage) {
-            return 1;
-        } else if(secondResultsPage){
-            return 2;
-        } else if (startPage === 0){
-            return Number(apiPage)*2;
+
+    const setPageLimit = (resultsPageNumber) => {
+        return resultsPageNumber >= MAX_PAGE;
+    };
+
+    const isOdd = (number) => number % 2 !== 0;
+    const startPage = (page) => {
+        if(searchState.primaryFilter === SEARCH_ALL){
+            return isOdd(page) ? 0 : 4;
+        } else {
+            return 0;
+        }
+
+    };
```

Comment:
const startPage = (page) => searchState.primaryFilter === SEARCH_ALL && !isOdd(page) ? 4 : 0;

Also consider making page size a variable, if that might be changed in the future (the `4`)